File: kotlin/bc-lifehash/src/main/kotlin/com/blockchaincommons/bclifehash/Patterns.kt
package com.blockchaincommons.bclifehash

/** The symmetry pattern used when rendering a LifeHash image. */
internal enum class Pattern {
    Snowflake,
    Pinwheel,
    Fiducial,
}

/** Selects a [Pattern] based on the rendering [version] and available [entropy]. */
internal fun selectPattern(entropy: BitEnumerator, version: Version): Pattern =
    when (version) {
        Version.Fiducial,
        Version.GrayscaleFiducial,
        -> Pattern.Fiducial

        else -> if (entropy.next()) Pattern.Snowflake else Pattern.Pinwheel
    }

File: typescript/bc-lifehash/src/patterns.ts
import { BitEnumerator } from './bit-enumerator.js';
import { Version } from './version.js';

export enum Pattern {
    Snowflake = 'snowflake',
    Pinwheel = 'pinwheel',
    Fiducial = 'fiducial',
}

export function selectPattern(
    entropy: BitEnumerator,
    version: Version,
): Pattern {
    if (
        version === Version.Fiducial ||
        version === Version.GrayscaleFiducial
    ) {
        return Pattern.Fiducial;
    }
    return entropy.next() ? Pattern.Snowflake : Pattern.Pinwheel;
}
